Diffing_Engine/Query/ObjectDifferences.cs

```diff
@@ -0,0 +1,170 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Diffing;
+using BH.oM.Base;
+using kellerman = KellermanSoftware.CompareNetObjects;
+using System.Reflection;
+using BH.Engine.Base;
+
+namespace BH.Engine.Diffing
+{
+    public static partial class Query
+    {
+        [Description("Compare two versions of the same object that sustained some modification over time, and returns differences.")]
+        [Input("pastObject", "Past version of the object (created before `followingObject`).")]
+        [Input("followingObject", "Following version of the object (modified or created after `pastObject`).")]
+        [Input("comparisonConfig", "Additional configurations to be used for the comparison.")]
+        [Output("Returns an `ObjectDifferences` object storing all the found differences between `previousObject` and `followingObject`." +
+            "\nIf no difference was found, returns null.")]
+        [PreviousVersion("5.0", "BH.Engine.Diffing.DifferentProperties(System.Object, System.Object, BH.oM.Base.ComparisonConfig)")]
+        public static ObjectDifferences ObjectDifferences(this object pastObject, object followingObject, BaseComparisonConfig comparisonConfig = null)
+        {
+            // Set ComparisonConfig if null. Clone it for immutability in the UI.
+            BaseComparisonConfig cc = comparisonConfig == null ? new ComparisonConfig() : comparisonConfig.DeepClone();
```

Comment:
Might need some specific null handling on `pastObject` and `followingObject` here

Reply:
Added one for both objects `null`. db23c5ff2972acec88405153ed6d154148938a20

If only one is null, we still need to defer to Kellerman library method below in order to get the appropriate result.